x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_initializer.tsx

```diff
@@ -23,52 +23,60 @@ import {
 import { FormattedMessage } from '@kbn/i18n-react';
 import type { MlJob } from '@elastic/elasticsearch/lib/api/types';
 import type { TimeRangeBounds } from '@kbn/ml-time-buckets';
-import type { SingleMetricViewerServices } from '..';
+import type { SingleMetricViewerEmbeddableInput } from '..';
 import { SeriesControls } from '../../application/timeseriesexplorer/components/series_controls';
 import {
   APP_STATE_ACTION,
   type TimeseriesexplorerActionType,
 } from '../../application/timeseriesexplorer/timeseriesexplorer_constants';
+import type { SingleMetricViewerEmbeddableCustomInput } from '../types';
 
 export interface SingleMetricViewerInitializerProps {
   bounds: TimeRangeBounds;
   defaultTitle: string;
-  initialInput?: SingleMetricViewerServices;
+  initialInput?: Partial<SingleMetricViewerEmbeddableInput>;
   job: MlJob;
-  onCreate: (props: {
-    panelTitle: string;
-    functionDescription?: string;
-    selectedDetectorIndex: number;
-    selectedEntities: any;
-  }) => void;
+  onCreate: (props: Partial<SingleMetricViewerEmbeddableCustomInput>) => void;
   onCancel: () => void;
 }
 
 export const SingleMetricViewerInitializer: FC<SingleMetricViewerInitializerProps> = ({
-  bounds,
   defaultTitle,
+  bounds,
   initialInput,
   job,
   onCreate,
   onCancel,
 }) => {
+  const isNewJob = initialInput?.jobIds !== undefined && initialInput?.jobIds[0] !== job.job_id;
```

Comment:
It does not refer to a newly created job, so I guess it tracks if the job id input has been update? 
```suggestion
  const isJobUpdated = initialInput?.jobIds !== undefined && initialInput?.jobIds[0] !== job.job_id;
```